### PR TITLE
Remove cast for jose.fields.

### PR DESCRIPTION
**Pull request by atombrella**
https://github.com/certbot/certbot/pull/9073 references this.

@adferrand @bmw 

## Pull Request Checklist

- [x] If the change being made is to a [distributed component](https://certbot.eff.org/docs/contributing.html#code-components-and-layout), edit the `master` section of `certbot/CHANGELOG.md` to include a description of the change being made.
- [x] Add or update any documentation as needed to support the changes in this PR.
- [x] Include your name in `AUTHORS.md` if you like.

### CODE REVIEW

**Comment by alexzorin** [comment]
https://github.com/certbot/josepy/pull/127 fixes the error in the first version of this PR. 

If nobody objects, I think we should cut a new release of `josepy` and upgrade to it, rather than keeping the remaining casts around.

I'll raise this in the meeting tomorrow.

**Comment by bmw** [comment]
I just released josepy 1.13.0 containing https://github.com/certbot/josepy/pull/127.

**Comment by alexzorin** [comment]
Thanks Brad.

@atombrella I think the oldest constraints need to be regenerated too with `tools/pinning/oldest/repin.sh`.

**Comment by atombrella** [comment]
> @atombrella I think the oldest constraints need to be regenerated too with `tools/pinning/oldest/repin.sh`.

I did ponder about the odd `version >= 3.7 and version >= 3.6`. Anyway, both scripts have been run :)

@bmw When merging, please edit the commit message so it doesn't contain all the commit messages from the intermediate commits. They'll be in the PR history.

File: tools/requirements.txt [hunk 1]
```diff
@@ -7,192 +7,191 @@
...
+pyasn1==0.4.8; python_version >= "3.7"
+pycparser==2.21; python_version >= "3.7"
 pygithub==1.55; python_version >= "3.7"
 pygments==2.11.2; python_version >= "3.7"
 pyjwt==2.3.0; python_version >= "3.7"
 pylev==1.4.0; python_version >= "3.7" and python_version < "4.0"
-pylint==2.12.2; python_version >= "3.7" and python_full_version >= "3.6.2"
-pynacl==1.5.0; python_version >= "3.7" or python_version >= "3.7"
+pylint==2.4.4; python_version >= "3.7"
+pynacl==1.5.0; python_version >= "3.7"
```
Comment (alexzorin) [inline tools/requirements.txt:119]: The current CI failure is caused by `pylint` being downgraded here. Not sure why Poetry has decided to do this, looking into it.

Comment (atombrella) [reply tools/requirements.txt:119]: :+1: I see you opened #9229 to solve this.

Comment (alexzorin) [reply tools/requirements.txt:119]: That's merged now, so hopefully `repin.sh` shouldn't crash anymore.

Try merge `master`, recreate the venv and re-run the pinning scripts. For me, doing this doesn't result in pylint being downgraded, so 🤞 .

Comment (atombrella) [reply tools/requirements.txt]: Nor for me :raised_hand_with_fingers_splayed: Thanks.

**Review by alexzorin — APPROVED** [review]
Sweet!

**Pull request merged**